timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -374,33 +374,21 @@ void declare_potential(py::module &m) {
 
                 std::vector<unsigned long long> du_dx(N * D);
                 std::vector<double> du_dp(P);
-
-                // initialize to zero for the accumulator
```

Comment:
I didn't understand why we were zeroing these, since it seems like they're overwritten -- maybe it's no longer needed?

Reply:
Kind of partial to keeping them zero'd, its more defensive than risking not overwriting everything. Or at least a comment that explicitly makes it clear that you must overwrite all values

Reply:
Hmm, thinking about this a bit more, I'd almost prefer the opposite, i.e. functions assume that output pointers initially point to uninitialized memory unless explicitly documented otherwise. In this view, something like what `execute_device` does (assuming `u` is initialized and just adding my energy) would be the exception to the rule and documented.

I think this could make the code a bit more readable because unused initializations can give the wrong impression about what a function does (e.g. I didn't initially understand that `execute_host` overwrites its outputs). What do you think?

Reply:
There's two types of behavior that we should consider:

1) Accumulators such as forces, energies, and du_dls, where different potentials accumulate to the same buffer. These need to be initialized to zero for correctness. 

2) Non-accumulators that rely on the underlying code to set the contents.

Usually when I read code that doesn't initialize, I *assume* that its contents will be set directly by the underlying call using an assignment operator somewhere. If it initializes to 0, then I assume that the underlying code will be accumulated some how (eg. via += or atomicAdds)

Reply:
For cases where performance aren't important (which I think of anything not CUDA or called in a loop) setting to 0 is safer. You get sane 'bad' values rather than values that might seem valid. Definitely spent a bunch of time tracking down junk initial memory values thinking they were incorrect computation or clashes in my system.

@proteneer distinction is definitely important but I can't say I have the right mindset to notice that difference on a quick read of the code.

Reply:
@badisa what do you think about initializing with a junk value other than 0 as @proteneer suggested? E.g. f9d970c

Reply:
Looks good!